### ISSUE TITLE
[bug] no-leaked-event-listener false positive with custom event name

### ISSUE THREAD

**Issue by k-sav** [Rel1cx/eslint-react#1340]
### Describe the bug

[@eslint-react/web-api/no-leaked-event-listener](https://eslint-react.xyz/docs/rules/web-api-no-leaked-event-listener) seems to fail with a custom event name

### Passing Example
```
import React, { useEffect } from "react";
function MyComponent() {
  useEffect(() => {
    const handleClick = () => {
      console.log("clicked");
    };
    document.addEventListener("click", handleClick);
    return () => {
      document.removeEventListener("click", handleClick);
    };
  }, []);
  return null;
}
```

### Actual
<img width="1434" height="438" alt="Image" src="https://github.com/user-attachments/assets/22073094-43a9-41f7-b831-1c3808ade261" />

### Platform and versions

```Text
"typescript-eslint": "^8.48.1",
"@eslint-react/eslint-plugin": "^2.3.12",
```

### Stack trace

```Text

```
